python/dgl/transform.py

```diff
@@ -575,6 +575,45 @@ def partition_graph_with_halo(g, node_part, num_hops):
         subg_dict[i] = subg
     return subg_dict
 
+def metis_partition(g, k, num_hops=1):
+    '''
+    This is to partition a graph with Metis partitioning.
+
+    The partitioned graph is stored in DGLGraph. The DGLGraph has the `inner_node` node data
+    that indicates whether a node is inside a partition and has the `inner_edge` edge data
+    that indicates whether an edge is inside a partition. In addition, it has the `part_id`
+    node data that indicates the partition a node belongs to.
+
+    Parameters
+    ------------
+    g: DGLGraph
+        The graph to be partitioned
+
+    k: int
+        The number of partitions.
+
+    num_hops: int
+        The number of hops a HALO node can be accessed.
+
+    Returns
+    --------
+    a dict of DGLGraphs
+        The key is the partition Id and the value is the DGLGraph of the partition.
+    '''
+    # METIS works only on symmetric graphs.
+    g = to_bidirected(g, readonly=True)
+    node_part = _CAPI_DGLMetisPartition(g._graph, k)
+    if len(node_part) == 0:
+        return None
+
+    node_part = utils.toindex(node_part)
+    parts = partition_graph_with_halo(g, node_part, num_hops)
+    node_part = node_part.tousertensor()
+    for part_id in parts:
+        part = parts[part_id]
+        part.ndata['part_id'] = F.gather_row(node_part, part.parent_nid)
+    return parts
```

Comment:
In Tzy's code, if the head node and tail node belong to different partitions, the edge will be assigned to both of these two partitions. Do we have the same behavior here?

Reply:
it only includes in-edges. I can change its behavior. Isn't Tzy's implementation replicates edges?